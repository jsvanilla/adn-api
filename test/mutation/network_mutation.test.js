const dataForTest = require('../dataForTest')
const testServer = require('../testServer')

const MUTATION_URL = '/mutation'

describe("Test the mutation enpoint",() =>{
  it("This adn chain has a mutation", done => {
    testServer().post(MUTATION_URL).send(dataForTest.mutationTrue).then(response => {
      expect(response.statusCode).toBe(200)
      done()
    })
  })
  it("This adn chain dont have a mutation", done => {
    testServer().post(MUTATION_URL).send(dataForTest.mutationFalse).then(response => {
      expect(response.statusCode).toBe(403)
      done()
    })
  })
  it("Check five chain", done => {
    testServer().post(MUTATION_URL).send(dataForTest.fiveChainMutation).then(response => {
      expect(response.statusCode).toBe(501)
      done()
    })
  })
  it("Check seven chain", done => {
    testServer().post(MUTATION_URL).send(dataForTest.sevenChainMutation).then(response => {
      expect(response.statusCode).toBe(501)
      done()
    })
  })
  it("Check lowecase character chain", done => {
    testServer().post(MUTATION_URL).send(dataForTest.dnaWithMinusCharacter).then(response => {
      expect(response.statusCode).toBe(501)
      done()
    })
  })
  it("Check a chain with out a character", done => {
    testServer().post(MUTATION_URL).send(dataForTest.dnaWithAnonCharacter).then(response => {
      expect(response.statusCode).toBe(501)
      done()
    })
  })
  it("Check chain with stange character", done => {
    testServer().post(MUTATION_URL).send(dataForTest.dnaWithStrangeCharacter).then(response => {
      expect(response.statusCode).toBe(501)
      done()
    })
  })
})

describe("Test the response snapshot", () =>{
  it("Check Mutation response Snapshot", done => {
    testServer().post(MUTATION_URL).send(dataForTest.mutationTrue).then(response => {
      const text = response.text
      expect(JSON.stringify(text)).toMatchSnapshot()
      done()
    })
  })
  it("Check Not mutation response Snapshot", done => {
    testServer().post(MUTATION_URL).send(dataForTest.mutationFalse).then(response => {
      const text = response.text
      expect(JSON.stringify(text)).toMatchSnapshot()
      done()
    })
  })
  it("Check Error Mutation response Snapshot", done => {
    testServer().post(MUTATION_URL).send(dataForTest.mutationFalse).then(response => {
      const text = response.text
      expect(JSON.stringify(text)).toMatchSnapshot()
      done()
    })
  })
})