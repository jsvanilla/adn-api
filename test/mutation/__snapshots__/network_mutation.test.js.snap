// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test the response snapshot Check Error Mutation response Snapshot 1`] = `"\\"{\\\\\\"error\\\\\\":false,\\\\\\"status\\\\\\":403,\\\\\\"body\\\\\\":\\\\\\"No se encontró mutación alguna en la cadena de ADN\\\\\\"}\\""`;

exports[`Test the response snapshot Check Mutation response Snapshot 1`] = `"\\"{\\\\\\"error\\\\\\":false,\\\\\\"status\\\\\\":200,\\\\\\"body\\\\\\":\\\\\\"¡ Se encontró una mutación en la cadena de ADN !\\\\\\"}\\""`;

exports[`Test the response snapshot Check Not mutation response Snapshot 1`] = `"\\"{\\\\\\"error\\\\\\":false,\\\\\\"status\\\\\\":403,\\\\\\"body\\\\\\":\\\\\\"No se encontró mutación alguna en la cadena de ADN\\\\\\"}\\""`;
